reference/troubleshooting.md

```diff
@@ -46,17 +46,181 @@ throughout the course of an update.
+
+To get a JSON representation of your stack's current state, you can export your current stack
+to a file:
+
+```bash
+$ pulumi stack export --file state.json
+```
+
+This file contains a lot of information. At the top-level, this JSON object has two fields:
+
+1. A `version`, indicating the version of the file format you're currently looking at. Don't
+change this.
+1. A `deployment`, which represents the state of the last deployment that this stack completed.
+
+The `deployment` object itself has three fields:
+
+1. A `manifest`, which contains some metadata about the previous deployment. You should not ever
+need to edit this.
+1. A list of `pending_operations`, which is a record of the operations that the Pulumi engine
+started but hasn't seen finish yet.
+1. A list of `resources`, which is a record of all resource that Pulumi knows about. When you create
+a resource, that resource's information is stored here.
+
+The possible fields of a resource are:
+
+1. `urn` - This resource's URN, or "universal resource name", which is a Pulumi-specific universal
+resource identifier.
+1. `custom` - A boolean indicating whether or not this resource is a "custom" resource, which means that
+it uses a resource provider to operate. Component resources are not `custom`.
+1. `delete` - A boolean indicating whether or not this resource is pending deletion.
+1. `id` - This resource's ID, which is a provider-specific resource identifier. This often corresponds to
+a cloud provider's identifier for a resource.
+1. `type` - The Pulumi type of this resource.
+1. `inputs` - A map of "inputs" for this resource. Inputs are the set of key-value pairs used as an input
+to a resource provider that created or updated the given resource.
+1. `outputs` - A map of "outputs" for this resource. Outputs are the set of key-value pairs that were given
+to Pulumi by a resource provider after a resource has been provisioned.
+1. `parent` - A URN for this resource's parent resource.
+1. `protect` - A boolean indicating whether or not this resource is protected. If a resource is protected,
+it can't be deleted.
+1. `external` - A boolean indicating whether or not this resource is "external" to Pulumi. If a resource is
+External, Pulumi does not own its life cycle and it will not ever delete or update the resource. Resources
+that are "read" using the `get` function are External.
+1. `dependencies` - A list of URNs indicating the resources that this resource depends on. Pulumi tracks dependencies
+between resources and so it is important that this list be the full list of resources upon which this resource
+depends.
+1. `initErrors` - A list of errors that occured that prevented this particular resource from initializing. Some resource
+providers (most notably Kubernetes) populate this field to indicate that a resource was created but failed to initialize.
+1. `provider` - A provider reference to the provider that is responsible for this particular resource.
```

Comment:
It's nice to have all this documented, I actually wonder if we want the core reference docs to include these details (and then reference them from here for what kind of edits to make for various troubleshooting scenarios).  Probably fine to tackle that in separate PR.

Reply:
I'm torn. If we expect users to sometimes have to edit their snapshot, I think we need to at least document it somewhere. If we don't, I'd be fine not doing so. Terraform does not do this.